Modules/posixmodule.c

```diff
@@ -79,6 +79,26 @@ corresponding Unix manual entries for more information on calls.");
 #  include <sys/stat.h>
 #endif /* HAVE_SYS_STAT_H */
 
+#if defined(__APPLE__)
+#include <TargetConditionals.h>
+#endif
+
+#if defined(__APPLE__) && HAVE_BUILTIN_AVAILABLE && !(TARGET_OS_OSX && __arm64__)
+#define HAVE_UTIMENSAT_RUNTIME __builtin_available(macos 10.13, ios 11, tvos 11, watchos 4, *)
+#define HAVE_FUTIMENS_RUNTIME  __builtin_available(macos 10.13, ios 11, tvos 11, watchos 4, *)
+#define HAVE_PREADV_RUNTIME    __builtin_available(macos 10.16, ios 14, tvos 14, watchos 7, *)
+#define HAVE_PWRITEV_RUNTIME    __builtin_available(macos 10.16, ios 14, tvos 14, watchos 7, *)
```

Comment:
Curious, what is the proper way to detect Big Sur, `10.16` or `11`?

Reply:
My understanding is that on current macs (with the Intel processor), upgrading to Big Sur will result in 10.16. However, on Apple Silicon, Big Sur will be 11.0.

Reply:
It's 11.0 for both Intel and Arm, however before the public disclosure it was marked as 10.16, so 10.16 appears in some of the headers like `uio.h`, and will be recognized by `_builtin_available`.   I just copied 10.16 out of `uio.h`,  but it's confusing so I'll change it to 11.0.